### PR TITLE
releases: switch to a soft release model

### PR DESCRIPTION
**Pull request by Stebalien**
Reworked https://github.com/ipfs/go-ipfs/pull/6496 based on feedback.

### CODE REVIEW

File: docs/releases.md [hunk 1]
```diff
@@ -9,28 +9,49 @@
...
+
+### Stage 2 - Public Beta
+
+At this stage, we'll announce the impending release to the community and ask for beta testers.
+
+**Goal:** Test the release in as many non-production environments as possible. This is relatively low-risk but gives us a _breadth_ of testing internal testing can't.
+
+### Stage 3 - Soft Release
+
+At this stage, we consider the release to be "production ready" and ask will ask the community and our early testers to (partially) deploy the release to their production infrastructure.
```
Comment (Stebalien) [inline docs/releases.md]: In https://github.com/ipfs/go-ipfs/pull/6496, I had a separate stages for asking early testers to test in production and asking the rest of the community. I removed that here to try to keep this from dragging on but can add it back.

Thoughts @daviddias, @lanzafame?

File: docs/releases.md [hunk 2]
```diff
@@ -54,91 +75,63 @@ The first step is for the `Lead Maintainer` for `go-ipfs` to open an issue with
...
-- Join the [Go Core Dev Team Weekly Sync](https://github.com/ipfs/team-mgmt/issues/650) and be part of the Sprint action!
-
-# ⁉️ Do you have questions?
-
-The best place to ask your questions about IPFS, how it works and what you can do with it is at [discuss.ipfs.io](http://discuss.ipfs.io). We are also available at the `#ipfs` channel on Freenode, which is also [accessible through our Matrix bridge](https://riot.im/app/#/room/#freenode_#ipfs:matrix.org).
+
+- version string in `version.go` has been updated
+- tag commit with vX.Y.Z-rcN
+
+- [ ] Stage 1 - Internal Testing
```
Comment (Stebalien) [inline docs/releases.md]: Tried to make it easier to track which stages have been completed (checking all the boxes _inside_ the stage doesn't mean that the stage itself is done).

File: docs/RELEASE_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+- Join an IPFS All Hands, introduce yourself and let us know where you would like to contribute - https://github.com/ipfs/team-mgmt/#weekly-ipfs-all-hands
+- Hack with IPFS and show us what you made! The All Hands call is also the perfect venue for demos, join in and show us what you built
+- Join the discussion at http://discuss.ipfs.io/ and help users finding their answers.
+- Join the [Go Core Dev Team Weekly Sync](https://github.com/ipfs/team-mgmt/issues/650) and be part of the Sprint action!
+
+### ⁉️ Do you have questions?
+
+The best place to ask your questions about IPFS, how it works and what you can do with it is at [discuss.ipfs.io](http://discuss.ipfs.io). We are also available at the `#ipfs` channel on Freenode, which is also [accessible through our Matrix bridge](https://riot.im/app/#/room/#freenode_#ipfs:matrix.org).
+
+### Changelog
```
Comment (Stebalien) [inline docs/RELEASE_TEMPLATE.md]: @daviddias see https://github.com/ipfs/go-ipfs/pull/6496#discussion_r302691789.

**Comment by daviddias** [comment]
Thank you for the review, @Stebalien. Going to merge this branch into the other one so that we can continue commenting on only one PR

**Pull request merged**